archunit/src/main/java/com/tngtech/archunit/core/domain/Formatters.java

```diff
@@ -122,4 +123,26 @@ public static String ensureSimpleName(String name) {
         return "";
     }
 
+    /**
+     * Returns the canonical array type name of any array type name passed in. Otherwise
+     * returns the passed type name as is. For example {@code [Ljava.lang.String;} will
+     * be reformatted to {@code java.lang.String[]} or {@code [I} will be reformatted
+     * to {@code int[]}, while {@code java.lang.String} would simply be returned as is.
+     * @param typeName A Java type name
+     * @return the passed type name, but for array type names the canonical array type name
+     */
+    @PublicAPI(usage = ACCESS)
+    public static String ensureCanonicalArrayTypeName(String typeName) {
+        if (!typeName.startsWith("[")) {
```

Comment:
I'll extract it to a function, but I think we shouldn't unify it with `JavaClassDescriptor.From.isArray`, because that method is more widely used and supports array class names as well as canonical names. But if we already have a canonical name, we don't really have to do anything. I mean, it wouldn't hurt, but I feel the overhead and "publification" of the private method `isArray` is not worth it :man_shrugging: Let's follow the rule of three, if we run into it a 3rd time we'll reconsider :wink: